sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -192,39 +196,214 @@ private void sendBody(HttpRequest httpRequest, Object progressReporter, HttpURLC
      *
      * @param httpRequest The HTTP Request being sent
      * @param connection The HttpURLConnection being sent to
-     *
      * @return A HttpResponse object
      */
     private HttpResponse receiveResponse(HttpRequest httpRequest, HttpURLConnection connection) {
         try {
             int responseCode = connection.getResponseCode();
+            Headers responseHeaders = getResponseHeaders(connection);
+            ServerSentEventListener listener = httpRequest.getServerSentEventListener();
+            RetrySSEResult retrySSEResult;
+            if (connection.getErrorStream() == null) {
+                if (isTextEventStream(responseHeaders) && listener != null) {
+                    try {
+                        BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+                        retrySSEResult = processBuffer(reader, listener);
+                        if (retrySSEResult != null) {
+                            retryExceptionForSSE(retrySSEResult, listener, httpRequest);
+                        }
+                    } catch (IOException e) {
+                        throw LOGGER.logThrowableAsError(new RuntimeException(e));
+                    }
+                }
+                return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders);
+            } else {
+                AccessibleByteArrayOutputStream outputStream = getAccessibleByteArrayOutputStream(connection);
+                return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders,
+                    BinaryData.fromByteBuffer(outputStream.toByteBuffer()));
+            }
+        } catch (IOException e) {
+            throw LOGGER.logThrowableAsError(new RuntimeException(e));
+        } finally {
+            connection.disconnect();
+        }
+    }
+
+    private Headers getResponseHeaders(HttpURLConnection connection) {
+        Map<String, List<String>> hucHeaders = connection.getHeaderFields();
+        Headers responseHeaders = new Headers((int) (hucHeaders.size() / 0.75F));
+        for (Map.Entry<String, List<String>> entry : connection.getHeaderFields().entrySet()) {
+            if (entry.getKey() != null) {
+                responseHeaders.add(HeaderName.fromString(entry.getKey()), entry.getValue());
+            }
+        }
+        return responseHeaders;
+    }
```

Comment:
This feels like it should be API on Headers, like we already have `Headers(Map<String, String>)`, we should have `Headers(Map<String, List<String>)`